src/Castle.Core/DynamicProxy/ProxyGenerationOptions.cs

```diff
@@ -204,7 +208,53 @@ public override int GetHashCode()
 			result = 29*result + (Selector != null ? 1 : 0);
 			result = 29*result + MixinData.GetHashCode();
 			result = 29*result + (BaseTypeForInterfaceProxy != null ? BaseTypeForInterfaceProxy.GetHashCode() : 0);
+		    result = 29*result + GetContentsHashCode(AdditionalAttributes);
 			return result;
 		}
+
+        private static int GetContentsHashCode<T>(IList<T> list)

```

Comment:
https://github.com/castleproject/Core/blob/master/src/Castle.Core/Core/Internal/CollectionExtensions.cs looks like it might be a reasonable place to put it, although I wouldn't make them extension methods probably.